antd/src/main/scala/demo/App.scala

```diff
@@ -410,7 +415,7 @@ object CSS extends js.Any
 
     val renderTooltip = section(
       h2("Tooltip"),
-      Tooltip.TooltipPropsWithTitleRefAttributes(title = ("Tooltip": ReactElement))(span("Hover me"))
+      Tooltip(TooltipPropsWithTitle("Tooltip": ReactElement).asInstanceOf[TooltipProps with RefAttributes[Any]])(span("Hover me"))
```

Comment:
It's a bit complicated. when the react components are generated sometimes we find components where the type of the props is a union type. two distinct set of valid props. the only way to encode that is to generate two components, so nesting is used for that. For instance `ToolTip.PropsA` .
Then there is the naming. typically these types are not explicitly named in typescript code, but we need them to be in this encoding. `TooltipPropsWithTitleRefAttributes` is likely derived from some combination of `TooltipProps`, `WithTitle`, `RefAttributes`. As you can imagine, when the converter comes up with these names it is consistent for the same code, but brittle across refactoring of the typescript code.

Reply:
re: better way. I guess the underlying problem is that the generated component havent gotten correct props so you need to specify the whole props object, right?
You can explicitly create both and combine them to avoid the cast, `TooltipProps(...).combineWith(RefAttributes[Any](...))`